### ISSUE TITLE
Inconsistent calls between fitgeom and fitsad

### ISSUE THREAD

**Issue by andrechalom** [piklprado/sads#68]
Calls to `fitsad` introduce truncation for some distributions; however the calls to the underlying `fitgeom`, etc, do not. This makes fits with `fitsad(..., sad="geom")` inconsistent with `fitgeom(...)`: 

```R
> fitgeom(moths)
Maximum likelihood estimation
Type: discrete species abundance distribution
Species: 240 individuals: 15609 
Call:
mle2(minuslogl = function (prob) 
-sum(dgeom(x, prob, log = TRUE)), start = list(prob = 0.0151429112246829), 
    method = "Brent", data = list(x = list(1, 1, 1, 1, 1, "etc")), 
    lower = 0, upper = 1)
Coefficients:
      prob 
0.01514291 
Log-likelihood: -1243.83 



> fitsad(moths, sad="geom")
Maximum likelihood estimation
Type: discrete species abundance distribution
Species: 240 individuals: 15609 
Call:
mle2(minuslogl = function (prob) 
-sum(dtrunc("geom", x = x, coef = prob, trunc = trunc, log = TRUE)), 
    start = list(prob = 0.0151429112246829), method = "Brent", 
    data = list(x = list(1, 1, 1, 1, 1, "etc")), lower = 0, upper = 1)
Coefficients:
prob 
   1 
Truncation point: 0 
Log-likelihood: -Inf 
```

**Comment by piklprado** [comment]
Seems to be a problem with Brent's method, which was called by fitgeom. I removed method=Brent and thus the fit now uses the default optmization of mle2.  Seems working now, please check.

**Comment by piklprado** [comment]
Side note: this issue actually can be optmized for the most common use of dgeom, which is zero-truncation (the default). In this case the mle is 1/mean(x), and no optimization is necessary. Also, probably the analytical solution for mle for any truncation point can be found and included here.

**Comment by andrechalom** [comment]
Well, there are actually two issues here. The first one is that the Log-likelihood is being reported as -Inf, and this is problably Brent's fault :stuck_out_tongue_closed_eyes: 

The second one is that the `fitgeom` is not using the truncation, the `fitsad` is using it, and there is no documentation about this difference. Shouldn't they give the same result? Maybe we could move the "truncation as default" to inside the `fitgeom` and the likes.

**Comment by piklprado** [comment]
I agree, and the issue concerning Brent's methods seems a little odd to me, as this method is in general very robust. Probably it is not for functions that goes to Inf too fast, someting to check.

**Comment by piklprado** [comment]
Concerning the second issue, by dafeult `fitgeom` truncates at zero but no `fitsad`. This is because zero-truncation for models that include zero is the usual choice. This was explained in the help of `fitsad'. Still,  `fitsad(x, "geom")` with argument `trunc` omitted will fit a zero-truncated geometric, and should return the same as `fitgeom(x)`. My fix was to make this happen.

**Comment by piklprado** [comment]
Also, this issue made me realize that the code of `fitgeom`, `fitpoligog` and `fitnbinom` with `trunc=0` as default did not allow to fit the full (not truncated) models. With my fix this is achieved with `trunc=NULL`. I changed the man pages accordingly. But not sure if this is a proper solution.

**Comment by piklprado** [comment]
Fixed in PR #73
